eth-bytecode-db-extractors/job-queue/src/migration.rs

```diff
@@ -0,0 +1,133 @@
+use sea_orm_migration::{
+    prelude::*,
+    sea_orm::{Statement, TransactionTrait},
+};
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        let create_extension_pgcrypto = r#"
+            -- Needed for gen_random_uuid()
+            CREATE EXTENSION IF NOT EXISTS "pgcrypto";
+        "#;
+        let create_function_job_queue_set_modified_at = r#"
+            CREATE OR REPLACE FUNCTION _job_queue_set_modified_at()
+                RETURNS TRIGGER AS
+            $$
+            BEGIN
+                NEW.modified_at = now();
+                RETURN NEW;
+            END;
+            $$ LANGUAGE plpgsql;
+        "#;
+        let create_enum_job_status = r#"
+            CREATE TYPE "_job_status" AS ENUM (
+                'waiting',
+                'in_process',
+                'success',
+                'error'
+            );
+        "#;
+        let create_table_job_queue = r#"
+            CREATE TABLE "_job_queue" (
+                "id" uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+
+                "created_at" timestamp NOT NULL DEFAULT (now()),
+                "modified_at" timestamp NOT NULL DEFAULT (now()),
+
+                "status" _job_status NOT NULL DEFAULT 'waiting',
+                "log" varchar
+            );
+        "#;
+        let create_index_job_queue_status =
+            "CREATE INDEX _job_queue_status_index ON _job_queue (status);";
+        let create_trigger_job_queue_set_modified_at = r#"
+            CREATE TRIGGER "set_modified_at"
+            BEFORE UPDATE ON "_job_queue"
+                FOR EACH ROW
+            EXECUTE FUNCTION _job_queue_set_modified_at();
+        "#;
+        let create_function_insert_job = r#"
+            CREATE OR REPLACE FUNCTION _insert_job()
+            RETURNS TRIGGER AS $$
+            BEGIN
+              -- Insert a new row into the jobs table and get the ID
+              INSERT INTO _job_queue DEFAULT VALUES
+              RETURNING id INTO NEW._job_id;
+
+              -- Update the jobs_id in the contract_addresses table
+              RETURN NEW;
+            END;
+            $$ LANGUAGE plpgsql;
+        "#;
+
+        from_statements(
+            manager,
+            &[
+                create_extension_pgcrypto,
+                create_function_job_queue_set_modified_at,
+                create_enum_job_status,
+                create_table_job_queue,
+                create_index_job_queue_status,
+                create_trigger_job_queue_set_modified_at,
+                create_function_insert_job,
```

Comment:
tbh, I don't like the idea that you declare 7 variables and then copypaste them here. if you miss some variable or write it twice, compiler wont tell you and it will be hard to spot the mistake. 

Maybe write them inside `&[]` and add `-- sql comment` if you need some kind of names for your sql statements

Reply:
same problem with `fn down()`

Reply:
oh, I see that you use the same approach in `blockscout-migration`... okay, then we can leave it as it is for consistency :)

Reply:
The problem here is that you cannot put a single string with several sql statements into `Statement::from_string`. Usually we are using `from_sql` method for that cases that internally split all statements based on ';'. But in functions we use ";" internally and thus, the approach with splitting cannot be applied here. That is why we need a separate statements